src/axes/axis.ts

```diff
@@ -150,9 +168,227 @@ export class Axis<D> extends Component {
+      case "right":
+        secondaryPosition = 0;
+        break;
+      case "top":
+        secondaryPosition = this.height();
+        break;
+      case "left":
+        secondaryPosition = this.width();
+        break;
+    }
+
+    let isHorizontal = this._isHorizontal();
+    bindElements(this._annotationContainer.select(".annotation-line-container"), "line", "annotation-line")
+      .attr({
+        x1: isHorizontal ? positionF : secondaryPosition,
+        x2: isHorizontal ? positionF : offsetF,
+        y1: isHorizontal ? secondaryPosition : positionF,
+        y2: isHorizontal ? offsetF : positionF,
+        visibility: visibilityF
+      });
+
+    bindElements(this._annotationContainer.select(".annotation-circle-container"), "circle", "annotation-circle")
+      .attr({
+        cx: isHorizontal ? positionF : secondaryPosition,
+        cy: isHorizontal ? secondaryPosition : positionF,
+        r: 3,
+        visibility: visibilityF
+      });
+
+    let rectangleOffsetF = (d: D) => {
+      switch (this.orientation()) {
+        case "bottom":
+        case "right":
+          return offsetF(d);
+        case "top":
+        case "left":
+          return offsetF(d) - measurements.get(d).height;
+      }
+    };
+    bindElements(this._annotationContainer.select(".annotation-rect-container"), "rect", "annotation-rect")
+      .attr({
+        x: isHorizontal ? positionF : rectangleOffsetF,
+        y: isHorizontal ? rectangleOffsetF : positionF,
+        width: isHorizontal ? (d) => measurements.get(d).width : (d) => measurements.get(d).height,
+        height: isHorizontal ? (d) => measurements.get(d).height : (d) => measurements.get(d).width,
+        visibility: visibilityF
+      });
+
+    let annotationWriter = this._annotationWriter;
+    let annotationFormatter = this.annotationFormatter();
+    bindElements(this._annotationContainer.select(".annotation-label-container"), "g", "annotation-label")
+      .attr({
+        transform: (d) => {
+          let xTranslate = isHorizontal ? positionF(d) : rectangleOffsetF(d);
+          let yTranslate = isHorizontal ? rectangleOffsetF(d) : positionF(d);
+          return "translate(" + xTranslate + "," + yTranslate + ")";
+        },
+        visibility: visibilityF
+      })
+      .each(function (annotationLabel) {
+        let writeOptions = {
+          selection: d3.select(this),
+          xAlign: "center",
+          yAlign: "center",
+          textRotation: isHorizontal ? 0 : 90
+        };
+        annotationWriter.write(annotationFormatter(annotationLabel),
+                                 isHorizontal ? measurements.get(annotationLabel).width : measurements.get(annotationLabel).height,
+                                 isHorizontal ? measurements.get(annotationLabel).height : measurements.get(annotationLabel).width,
+                                 writeOptions);
+      });
+  }
+
+  protected _annotatedTicksInDomain() {
+    return this.annotatedTicks();
```

Comment:
Still works, since that function operates in pixel-space. Values not in the domain of `Scales.Category` get scaled to `NaN`, which won't be in the `range()`.